### PR TITLE
Hal API to retrieve TID stats via iw and Netlink

### PR DESCRIPTION
**Pull request by Rajakumaran1106**
Signed-off-by: Rajakumaran1106 <rajakumaran.a@ltts.com>

### CODE REVIEW

File: source/wifi/wifi_hal.c [hunk 1]
```diff
@@ -6700,10 +6722,175 @@ INT wifi_setRadioStatsEnable(INT radioIndex, BOOL enable)
...
+    return NL_SKIP;
+}
+#endif
+
 INT wifi_getApAssociatedDeviceTidStatsResult(INT radioIndex,  mac_address_t *clientMacAddress, wifi_associated_dev_tid_stats_t *tid_stats,  ULLONG *handle)
 {
-    // TODO Implement me!
-    return RETURN_ERR;
+#ifdef HAL_NETLINK_IMPL
+   Netlink nl;
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:6789]: Indent should be 4 spaces for this block. Here 3 spaces are given.

File: source/wifi/wifi_hal.c [hunk 2]
```diff
@@ -6700,10 +6722,175 @@ INT wifi_setRadioStatsEnable(INT radioIndex, BOOL enable)
...
+            stats_entry->num_msdus = atol(buf);
+
+        stats_entry->ac = _tid_ac_index_get[tid_index];
+//      TODO:
+//      ULLONG ewma_time_ms;    <! Moving average value based on last couple of transmitted msdus
+//      ULLONG sum_time_ms; <! Delta of cumulative msdus times over interval
+    }
+    return RETURN_OK;
+#endif
 }
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:6894]: I would like to have test caller from main function. Do we have provision for any one test this particular API?

File: source/wifi/wifi_hal.c [hunk 3]
```diff
@@ -6700,10 +6722,175 @@ INT wifi_setRadioStatsEnable(INT radioIndex, BOOL enable)
...
+            if(nla_parse_nested(stats_info, NL80211_TID_STATS_MAX,tidattr, tid_policy)) {
+                printf("failed to parse nested stats attributes!");
+                return;
+            }
+        }
+
+        if(stats_info[NL80211_TID_STATS_TX_MSDU])
+            stats_entry->num_msdus = (unsigned long long)nla_get_u64(stats_info[NL80211_TID_STATS_TX_MSDU]);
+
+            tid_index++;
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:6777]: we don't need extra 4 spaces for these 2 lines

File: source/wifi/wifi_hal.c [hunk 4]
```diff
@@ -6700,10 +6722,175 @@ INT wifi_setRadioStatsEnable(INT radioIndex, BOOL enable)
...
+    wifi_associated_dev_tid_entry_t *stats_entry;
+
+    snprintf(if_name,sizeof(if_name),"wlan%d",radioIndex);
+
+    strcpy(mac_addr,clientMacAddress);
+
+    snprintf(pipeCmd,sizeof(pipeCmd),"iw dev %s station dump -v > "TID_STATS_FILE,if_name);
+    fp= popen(pipeCmd,"r");
+
+    if(fp == NULL)
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:6855]: I would like to have this line 'perror("popen failed");' for all popen error condition

File: source/wifi/wifi_hal.c [hunk 5]
```diff
@@ -77,6 +77,28 @@
...
+    WIFI_RADIO_QUEUE_TYPE_BK,      /* 9 */
+    WIFI_RADIO_QUEUE_TYPE_BK,      /* 10 */
+    WIFI_RADIO_QUEUE_TYPE_BE,      /* 11 */
+    WIFI_RADIO_QUEUE_TYPE_VI,      /* 12 */
+    WIFI_RADIO_QUEUE_TYPE_VI,      /* 13 */
+    WIFI_RADIO_QUEUE_TYPE_VO,      /* 14 */
+    WIFI_RADIO_QUEUE_TYPE_VO,      /* 15 */
+};
+
+
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:101]: remove this line

**Review by manigandanj — CHANGES_REQUESTED** [review]

File: source/wifi/wifi_hal.c (hunk 2 above)
Comment (Rajakumaran1106) [reply source/wifi/wifi_hal.c:6894]: I can give you my test code, 

Place the below content into main

int radioIndex=0;
u64 handle= 0;
char sta2[] = "08:7f:98:97:c8:17";
ac_address_t *st2;
st2=sta2;

wifi_associated_dev_tid_stats_t tid_stats;
wifi_getApAssociatedDeviceTidStatsResult(radioIndex,st2,&tid_stats,handle);

File: source/wifi/wifi_hal.c (hunk 3 above)
Comment (Rajakumaran1106) [reply source/wifi/wifi_hal.c:6777]: Noted

File: source/wifi/wifi_hal.c (hunk 4 above)
Comment (Rajakumaran1106) [reply source/wifi/wifi_hal.c:6855]: Noted

File: source/wifi/wifi_hal.c (hunk 5 above)
Comment (Rajakumaran1106) [reply source/wifi/wifi_hal.c:101]: Noted

File: source/wifi/wifi_hal.c (hunk 1 above)
Comment (Rajakumaran1106) [reply source/wifi/wifi_hal.c:6789]: Noted

**Pull request closed without merging**